backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHIteratorApi.scala

```diff
@@ -87,7 +88,7 @@ class CHIteratorApi extends IIteratorApi with Logging {
     if (index < 3) {
       logDebug(s"The substrait plan for partition $index:\n${substraitPlan.toString}")
     }
-    NativePartition(index, substraitPlan.toByteArray, localFilesNodesWithLocations.head._2)
+    GlutenPartition(index, substraitPlan, localFilesNodesWithLocations.head._2)
```

Comment:
`GlutenPartition` will be serializated to each task, why include `Plan` object in GlutenPartition ? Is it better use `byte[]` and reduce data ?

Reply:
This is to preserve a structured object rather than bytes for JVM codes. Since technically it might be more reasonable to explicitly serialize the plan only when sending to backend 's native code.

Considering a backend that executes a plan in pure JVM code (or do some extra validations before sending to native side), the current `NativePartition` requires manually deserializing from byte[] to a Java plan object then to execute it. While preserving a Java object in partition may avoid that.

Regarding the data size I'll run TPC-H SF 1K to check if regressions are caused by the patch.

Reply:
Got it.